app/views/layouts/application.html.erb

```diff
@@ -1,26 +1,45 @@
 <!DOCTYPE html>
 <html>
-  <title><%= CONFIG["project_shortname"] %></title>
+  <title><% if !@title.blank? %><%= @title %> | <% end %><%= CONFIG["project_shortname"] %></title>
+  <meta name="application-name" content="<%= CONFIG["project_shortname"] %>">
+  <% if !@description.blank? %>
+  <meta name="description" content="<%= @description %>">
+  <% end %>
 
+  <meta name="viewport" content="width=device-width, initial-scale=1">
   <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
   <meta charset="utf-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
   <%= csrf_meta_tags %>
-  <%# favicon_link_tag %>
+
+  <!-- Favicons -->
+  <%#= favicon_link_tag "favicon.png", rel: "icon", sizes: "192x192", type: nil %>
+  <%#= favicon_link_tag "apple-touch-icon.png", rel: "apple-touch-icon", sizes: "180x180", type: nil %>
+  <!-- <meta name="msapplication-config" content="<%= browserconfig_path format: :xml %>"> -->
+  <%#= favicon_link_tag "favicon.ico", rel: "icon", type: nil %>
 
   <!-- Stylesheets -->
   <%= stylesheet_link_tag 'application', media: 'all' %>
+  <% if !@extra_css.blank? %>
+    <% Array.wrap(@extra_css).each do |css|
+      concat "  #{css}".html_safe
```

Comment:
same here as with the javascript, I feel like it's a bit cleaner to have the `stylesheet_link_tag` here instead of in the controller